crates/algokit_utils/src/transactions/composer.rs

```diff
@@ -494,10 +738,34 @@ impl Composer {
             .try_for_each(|transaction| self.add_transaction(transaction))
     }
 
+    pub fn add_transaction_with_signer(
+        &mut self,
+        transaction: TransactionWithSigner,
+    ) -> Result<(), ComposerError> {
+        self.push(ComposerTransaction::TransactionWithSigner(transaction))
+    }
+
+    pub fn add_transactions_with_signers(
+        &mut self,
+        transactions: Vec<TransactionWithSigner>,
+    ) -> Result<(), ComposerError> {
+        if self.transactions.len() + transactions.len() > MAX_TX_GROUP_SIZE {
+            return Err(ComposerError::GroupSizeError());
+        }
+
+        transactions
+            .into_iter()
+            .try_for_each(|transaction| self.add_transaction_with_signer(transaction))
+    }
+
     pub fn transactions(&self) -> &Vec<ComposerTransaction> {
         &self.transactions
     }
 
+    pub fn signed_group(&self) -> Option<&Vec<SignedTransaction>> {
```

Comment:
We already have this functionality via gather_signatures and it accounts for if the group has been built. Is this something that utils-ts/py already has support for?